### PR TITLE
[Test][GEMM] Convert `f16_f16_f32` harness and tests

### PR DESCRIPTION
**Pull request by Aaron-Hutchinson**
This converts the widening f16 GEMM harness into the new format from #56. Tests for three kernels are added using the new harness:

- skl_gemm_f16_f16_f32_zvfh_x390
- skl_gemm_a1b01_f16c_f16_f32_xsfmm32a16f
- skl_gemm_a1b01_f16pc_f16cp_f32rcp_xsfmm32a16f

### CODE REVIEW

File: test/gemm/xsfmm/skl_gemm_a1b01_f16c_f16_f32_xsfmm32a16f.c [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+#ifdef SKL_ENABLE_BENCHMARKS
+    // Benchmark tests
+    {BENCH, .m1 = 128, .n1 = 128, .k1 = 2048, .alpha = 1.f, .beta = 0.f},
+    {BENCH, .m1 = 128, .n1 = 128, .k1 = 2048, .alpha = 1.f, .beta = 1.f},
+#endif // SKL_ENABLE_BENCHMARKS
+
+#ifdef SKL_ENABLE_TESTS
+    // Verification tests - comprehensive coverage for Xsfmm A1B01 layout (ETE=64)
+    /* Edge case: 1x1 matrix with k=0 (no computation, C = beta * C) */
+    {TEST, .m1 = 1,   .n1 = 1,   .k1 = 0, .alpha = 1.f},
```
Comment (Aaron-Hutchinson) [inline test/gemm/xsfmm/skl_gemm_a1b01_f16c_f16_f32_xsfmm32a16f.c:67]: @myeh01 For the two xsfmm32a16f kernels I just copied the test suite from the f32 files. Please suggest any adjustments that should be made.

File: test/gemm/gemm_f16rcprc_f16rcprc_f32rcprc.c [hunk 2]
```diff
@@ -1,187 +1,132 @@
...
+        h->c_pack.len ? malloc(h->c_pack.len * sizeof(*(h->ctx.bound))) : NULL;
+    if (h->c_pack.len) {
+      memcpy(h->ctx.ref_c, h->c_pack.data,
+             h->c_pack.len * sizeof(*(h->c_pack.data)));
+    }
+  }
+}
+
+void gemm_f16rcprc_f16rcprc_f32rcprc_verify(skl_test_t *t) {
   /* Compute the reference matrix output. */
```
Comment (myeh01) [inline test/gemm/gemm_f16rcprc_f16rcprc_f32rcprc.c]:
```suggestion
```
I'll fix this in the f32 harness.

File: test/gemm/gemm_f16rcprc_f16rcprc_f32rcprc.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2026 SiFive, Inc.
```
Comment (myeh01) [inline test/gemm/gemm_f16rcprc_f16rcprc_f32rcprc.h]: Licenses will need to be updated.

File: test/gemm/xsfmm/skl_gemm_a1b01_f16pc_f16cp_f32rcp_xsfmm32a16f.c [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+    {TEST, .m1 = 7, .n1 = 1, .k1 = 1, .alpha = 1.f, .beta = 0.f},
+    {TEST, .m1 = 7, .n1 = 2, .k1 = 1, .alpha = 1.f, .beta = 0.f},
+    {TEST, .m1 = 7, .n1 = 3, .k1 = 1, .alpha = 1.f, .beta = 0.f},
+    {TEST, .m1 = 7, .n1 = 4, .k1 = 1, .alpha = 1.f, .beta = 0.f},
+    {TEST, .m1 = 7, .n1 = 5, .k1 = 1, .alpha = 1.f, .beta = 0.f},
+    {TEST, .m1 = 7, .n1 = 6, .k1 = 1, .alpha = 1.f, .beta = 0.f},
+    {TEST, .m1 = 7, .n1 = 7, .k1 = 1, .alpha = 1.f, .beta = 0.f},
+
+    {TEST, .m1 = 7, .n1 = 1, .k1 = 2, .alpha = 1.f, .beta = 0.f},
```
Comment (myeh01) [inline test/gemm/xsfmm/skl_gemm_a1b01_f16pc_f16cp_f32rcp_xsfmm32a16f.c:87]: Could the block of tests on lines 85-91 be replicated with `.k1 = 3`, `.k1 = 4`, and `.k1 = 5`?

Also, a more general question: how comprehensive do we want test suites to be? I added a lot of test cases for the Xsfmm kernels, but I'm not sure if it's excessive.

**Comment by myeh01** [comment]
LGTM! Just a few minor things.

File: test/gemm/xsfmm/skl_gemm_a1b01_f16pc_f16cp_f32rcp_xsfmm32a16f.c (hunk 4 above)
Comment (Aaron-Hutchinson) [reply test/gemm/xsfmm/skl_gemm_a1b01_f16pc_f16cp_f32rcp_xsfmm32a16f.c:87]: Thanks for the recommendation!

I think the only current limitation is test/CI time, but the more coverage the better. After we complete the test conversions we should probably assess the state of CI.

**Review by myeh01 — APPROVED** [review]
LGTM!
